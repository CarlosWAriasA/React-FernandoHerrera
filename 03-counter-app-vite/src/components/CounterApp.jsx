import React, { useState } from "react";
import PropTypes from "prop-types";

function CounterApp({ value }) {
  const [counter, setCounter] = useState(value);

  const handleAddition = () => {
    // setCounter(counter + 1);
    return setCounter((actualValue) => {
      return actualValue + 1;
    });
  };

  const handleSubtract = () => {
    return setCounter(counter - 1);
  };

  const handleReset = () => {
    return setCounter(value);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2> {counter} </h2>
      <button onClick={handleAddition}>+1</button>
      <button onClick={handleSubtract}>-1</button>
      <button aria-label="btn-reset" onClick={handleReset}>
        Reset
      </button>
    </>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};

export default CounterApp;
