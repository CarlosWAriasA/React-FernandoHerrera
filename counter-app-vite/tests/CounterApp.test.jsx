import { fireEvent, render, screen } from "@testing-library/react";
import CounterApp from "../src/components/CounterApp";
import React from "react";

describe("Pruebas en <CounterApp />", () => {
  const value = 100;
  test("should match with the snapshot", () => {
    const { container } = render(<CounterApp value={value} />);
    expect(container).toMatchSnapshot();
  });

  test("should show the default value of 100", () => {
    render(<CounterApp value={value} />);
    expect(screen.getByText(value)).toBeTruthy();
  });

  test("should increment with the +1 button", () => {
    render(<CounterApp value={value} />);
    fireEvent.click(screen.getByText("+1"));
    expect(screen.getByText(value + 1)).toBeTruthy();
  });

  test("should decrement with the -1 button", () => {
    render(<CounterApp value={value} />);
    fireEvent.click(screen.getByText("-1"));
    expect(screen.getByText(value - 1)).toBeTruthy();
  });

  test("should reset with the reset button", () => {
    render(<CounterApp value={value} />);
    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("+1"));
    fireEvent.click(screen.getByText("+1"));
    // fireEvent.click(screen.getByText("Reset"));
    fireEvent.click(screen.getByRole("button", { name: "btn-reset" }));
    expect(screen.getByText(value)).toBeTruthy();
  });
});
