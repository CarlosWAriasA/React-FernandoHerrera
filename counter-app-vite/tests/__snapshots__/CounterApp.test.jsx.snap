// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en <CounterApp /> should match with the snapshot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h2>
     
    100
     
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button
    aria-label="btn-reset"
  >
    Reset
  </button>
</div>
`;
